assignment/final/aLife/code/master.js

```diff
@@ -0,0 +1,417 @@
+        <div class="col-sm-6">${pkmn.STATS[1].STAT}: ${pkmn.STATS[1].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[2].STAT}: ${pkmn.STATS[2].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[3].STAT}: ${pkmn.STATS[3].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[4].STAT}: ${pkmn.STATS[4].VALUE}</div>
+        <div class="col-sm-6">${pkmn.STATS[5].STAT}: ${pkmn.STATS[5].VALUE}</div>
+        `;
+
+    //console.log(pkmn.MOVES);
+    for (let i = 0; i < pkmn.MOVES.length; i++) {
+        moveset.innerHTML +=
+            `
+            <div class="row">
+                <div class="col-sm-5 offset-1">${pkmn.MOVES[i].NAME}</div>
+                <div class="col-sm-2">${pkmn.MOVES[i].TYPE}</div>
+                <div class="col-sm-1" style="text-align: right;">${pkmn.MOVES[i].POWER != null ? pkmn.MOVES[i].POWER : ""}</div>
+            </div>
+            `;
+    }
+
+    idBox.innerHTML = pkmn.ID;
+}
+
+function filterPkmn() {
+    const regex = new RegExp(".*" + pkmnName.value + ".*");
+    let count = pkmnList.filter(pkmn => pkmn.NAME.match(regex)).length;
+    pkmnListDiv.innerHTML = "";
+    for (let i = 0; i < count; i++) {
+        pkmnListDiv.innerHTML += `<p id="${pkmnList.filter(pkmn => pkmn.NAME.match(regex))[i].NAME}">${
+            pkmnList.filter(pkmn => pkmn.NAME.match(regex))[i].NAME}</p>`;
+    }
+}
+
+function filterByType() {
+    let alreadyAdded = [];
+
+    for(let i = 0; i < typesCorespondingPkmn.length; i++)
+    {
+        if(typesCorespondingPkmn[i].TYPE.TYPE == selectedType1.toLowerCase() ||
+        typesCorespondingPkmn[i].TYPE.TYPE == (selectedType2 != null? selectedType2.toLowerCase(): "")){
+            pkmnListForTypeFilter.push(typesCorespondingPkmn[i]);
+        }
+    }
+
+    if(selectedType2 == null){
+        //console.log(pkmnListForTypeFilter);
+        pkmnListDiv.innerHTML = "";
+        for(let i = 0; i < pkmnListForTypeFilter[0].LIST.length; i++){
+            if(alreadyAdded.indexOf(makingThisEasier(i)) < 0){
+                pkmnListDiv.innerHTML +=
+                `
+                <p id="${makingThisEasier(i) != ""? makingThisEasier(i): Math.random()}">
+                ${makingThisEasier(i) != ""? makingThisEasier(i): "Some pkmn from Alolan region are not included."}
+                </p>
+                `
+                alreadyAdded.push(makingThisEasier(i));
+            }
+        }
+        alreadyAdded = [];
+    }
+}
+
+function makingThisEasier(i){
+    let subString = pkmnListForTypeFilter[0].LIST[i].pokemon.name.substring(0, pkmnListForTypeFilter[0].LIST[i].pokemon.name.indexOf("-"));
+    //console.log("Substring = " + subString);
+    try{
+        pkmn = pkmnList.filter(pkmn => pkmn.NAME == pkmnListForTypeFilter[0].LIST[i].pokemon.name)[0].NAME;
+        //console.log(pkmn);
+    }catch(err){
+        //console.log("Error from filter on " + pkmnListForTypeFilter[0].LIST[i].pokemon.name);
+        try{
```

Comment:
I know `try/catches` feel nice but usually they are kind of frowned on in JS.  Perhaps we can look at your code and see why these are necessary and potentially some ways to avoid them =)

Reply:
I’m unsure if it’s the code or the API, the reason for it is because the API has Pokémon in the types lists that don’t exist/are named differently in the national Pokédex I originally pulled info from. So when filtering by type those Pokémon show up but can’t be found in the list so it throws errors.
The part where it returns a string up to a certain character was so it would return things like giratina even though the name from they type list was giratina-altered.. This then caused problems with Pokémon that actually do have a - in their name though.